### PR TITLE
Protostar pagination update

### PR DESCRIPTION
**Pull request by brianteeman**
Improve thew accessibility of the pagination - there are no visual changes

1. Wrap the pagination in a nav and add a role (needed for IE) and an aria-label
2. Ensure font icons have aria-hidden=true
3. Add aria-current to the selected page
4. Add aria-title to the active links eg Go to page 1

### Note 1
that the disabled links will not be announced to a screen reader which is the intended behaviour

### Note 3
the list is an unordered list which appears to be the standard as otherwise the links would be announced as Link 1 go to page 2 etc which is nonsense

### Note 4
I dont know why but protostar uses an override for pagination instead of the layout. For b/c this PR is just for the override but when approved I will do a similar PR for the layout

Thanks to @fuzzbomb (Drupal 8 Core Accessibility Maintainer) for his advice and reviewing this

### CODE REVIEW

File: templates/protostar/html/pagination.php [hunk 1]
```diff
@@ -112,7 +112,8 @@ function pagination_list_render($list)
 		}
 	}
 
-	$html = '<ul class="pagination-list">';
+	$html  = '<nav role="navigation" aria-label="' . JText::_('JPAGINATION') . '">';
```
Comment (ciar4n) [inline templates/protostar/html/pagination.php]: JPAGINATION... is that correct? Can't find this constant anywhere.

Comment (C-Lodder) [reply templates/protostar/html/pagination.php]: Should be `JLIB_HTML_PAGINATION`

Comment (brianteeman) [reply templates/protostar/html/pagination.php]: Oops

File: templates/protostar/html/pagination.php [hunk 2]
```diff
@@ -151,25 +153,25 @@ function pagination_item_active(&$item)
...
 	{
-		$display = '<span class="icon-previous"></span>';
+		$display = '<span class="icon-previous" aria-hidden="true"></span>';
 	}
 
 	// Check for "Next" item
 	if ($item->text === JText::_('JNEXT'))
 	{
-		$display = '<span class="icon-next"></span>';
+		$display = '<span class="icon-next" aria-hidden="true" ></span>';
```
Comment (Quy) [inline templates/protostar/html/pagination.php]: Remove space `"true" >`

**Comment by ciar4n** [comment]
I have tested this item :white_check_mark: successfully on 08d06388651e5423669a1bcb6c3b2fc05df6f132<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18326">issues.joomla.org/tracker/joomla-cms/18326</a>.</sub>

**Comment by Quy** [comment]
I would prefer it to say `Go to next page` rather than `Go to page Next`. The same goes for `Start`, `Prev`, and `End`. Also should this change to use `sprintf`?

**Comment by brianteeman** [comment]
@quy I agree - the sprintf for the page number is easy - will need a little code change for the prev,next strings - i will look at it on sunday

**Comment by brianteeman** [comment]
@quy I have had a go at the sprintf stuff - it works but maybe you can optimise it and make it DRY

File: templates/protostar/html/pagination.php [hunk 3]
```diff
@@ -196,31 +204,31 @@ function pagination_item_inactive(&$item)
...
 	{
-		return '<li class="disabled"><a><span class="icon-last"></span></a></li>';
+		return '<li class="disabled"><a><span class="icon-last" aria-hidden="true"></span></a></li>';
 	}
 
 	// Check if the item is the active page
 	if (isset($item->active) && $item->active)
 	{
-		return '<li class="active hidden-phone"><a>' . $item->text . '</a></li>';
+		return '<li class="active hidden-phone"><a aria-current="true" aria-label="' . JText::_('JLIB_HTML_PAGE') . ' ' . $item->text . '">' . $item->text . '</a></li>';
```
Comment (Quy) [inline templates/protostar/html/pagination.php]: sprintf here? `JText::_('JLIB_HTML_PAGE') . ' ' . $item->text`

**Comment by Quy** [comment]
Looks good! 

`Go to prev page`  I understand why it is `prev`, but it would be nice to be `previous`.

**Comment by brianteeman** [comment]
Changing to previous will have to wait until this is approved and I can work n the pagination layout as well

File: templates/protostar/html/pagination.php (hunk 3 above)
Comment (brianteeman) [reply templates/protostar/html/pagination.php]: Done

File: administrator/language/en-GB/en-GB.lib_joomla.ini [hunk 4]
```diff
@@ -425,11 +425,16 @@ JLIB_HTML_ERROR_FUNCTION_NOT_SUPPORTED="Function not supported."
 JLIB_HTML_ERROR_NOTFOUNDINFILE="%s: :%s not found in file."
 JLIB_HTML_ERROR_NOTSUPPORTED_NOFILE="%s: :%s not supported. File not found."
 JLIB_HTML_ERROR_NOTSUPPORTED="%s: :%s not supported."
+JLIB_HTML_GOTO_PAGE="Go to page %s"
+JLIB_HTML_GOTO_POSITION="Go to %s page"
 JLIB_HTML_MOVE_DOWN="Move Down"
 JLIB_HTML_MOVE_UP="Move Up"
 JLIB_HTML_NO_PARAMETERS_FOR_THIS_ITEM="There are no parameters for this item."
 JLIB_HTML_NO_RECORDS_FOUND="No records found."
+JLIB_HTML_PAGE="Page"
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.lib_joomla.ini]: Delete

File: language/en-GB/en-GB.lib_joomla.ini [hunk 5]
```diff
@@ -425,11 +425,16 @@ JLIB_HTML_ERROR_FUNCTION_NOT_SUPPORTED="Function not supported."
 JLIB_HTML_ERROR_NOTFOUNDINFILE="%s: :%s not found in file."
 JLIB_HTML_ERROR_NOTSUPPORTED_NOFILE="%s: :%s not supported. File not found."
 JLIB_HTML_ERROR_NOTSUPPORTED="%s: :%s not supported."
+JLIB_HTML_GOTO_PAGE="Go to page %s"
+JLIB_HTML_GOTO_POSITION="Go to %s page"
 JLIB_HTML_MOVE_DOWN="Move Down"
 JLIB_HTML_MOVE_UP="Move Up"
 JLIB_HTML_NO_PARAMETERS_FOR_THIS_ITEM="There are no parameters for this item."
 JLIB_HTML_NO_RECORDS_FOUND="No records found."
+JLIB_HTML_PAGE="Page"
```
Comment (Quy) [inline language/en-GB/en-GB.lib_joomla.ini]: Delete

File: templates/protostar/html/pagination.php [hunk 6]
```diff
@@ -196,31 +204,33 @@ function pagination_item_inactive(&$item)
...
 	{
-		return '<li class="disabled"><a><span class="icon-last"></span></a></li>';
+		return '<li class="disabled"><a><span class="icon-last" aria-hidden="true"></span></a></li>';
 	}
 
 	// Check if the item is the active page
 	if (isset($item->active) && $item->active)
 	{
-		return '<li class="active hidden-phone"><a>' . $item->text . '</a></li>';
+		$aria    = JText::sprintf('JLIB_HTML_PAGE_CURRENT', $item->text);
```
Comment (Quy) [inline templates/protostar/html/pagination.php]: Remove extra spaces before `=`

File: administrator/language/en-GB/en-GB.lib_joomla.ini (hunk 4 above)
Comment (brianteeman) [reply administrator/language/en-GB/en-GB.lib_joomla.ini]: No - we dont delete strings ;)

Comment (brianteeman) [reply administrator/language/en-GB/en-GB.lib_joomla.ini]: ah whoops it was me that added it

Comment (Quy) [reply administrator/language/en-GB/en-GB.lib_joomla.ini]: Delete this one too

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on fa05cc6497dedfd33ea166ec9d436d84423cc032<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18326">issues.joomla.org/tracker/joomla-cms/18326</a>.</sub>

**Comment by brianteeman** [comment]
See #18357 for similar work on the ISIS template

**Comment by ghost** [comment]
@ciar4n can you please test again?

**Comment by C-Lodder** [comment]
I'm seeing a lot of `<a>` elements without `href` which I think may be an issue for a11y in regards to tab indexing.

@zwiastunsw ?

Could be wrong

**Comment by brianteeman** [comment]
It s perfectly ok because there is no link in there it is ignored. We checked that before submitting as we were not sure either

**Comment by brianteeman** [comment]
See the test report here https://gist.github.com/brianteeman/46e1cf36b6c57944c4b9c5efcad84c1d#gistcomment-2225785

**Comment by dgrammatiko** [comment]
> I'm seeing a lot of <a> elements without href which I think may be an issue for a11y in regards to tab indexing.

Just use `href="javascript:void(0)"`

**Comment by brianteeman** [comment]
There is no need

**Comment by zwiastunsw** [comment]
@brianteeman: Thanks to this work. 
In my opinion, no additional nav tag is needed. Just replace the nav tag div in function pagination_list_footer. (This footer is part of navigation).
If you think you want to stay, as you proposed, let us know. Or correct. 
And in one case and the other, I will test it successfully.  

@C-Lodder:
>I'm seeing a lot of <a> elements without href which I think may be an issue for a11y in regards to tab indexing.

It's OK.

**Comment by ciar4n** [comment]
I have tested this item :white_check_mark: successfully on fa05cc6497dedfd33ea166ec9d436d84423cc032<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18326">issues.joomla.org/tracker/joomla-cms/18326</a>.</sub>

**Comment by brianteeman** [comment]
The nav tag is absolutely required.and i will not be removing it. Every accessibility source clearly says that this is a requirement. I am not surprised that the a11y team once again chose to do nothing and instead criticise those that actually contribute to making joomla more accessible

**Comment by zwiastunsw** [comment]
1. I am sorry, but you misunderstood, of course, the nav tag is required. I have only proposed to include the whole element. Instead of div tag in function pagination_list_footer  use nav tag:
````php
function pagination_list_footer($list)
{
	$html = "<nav role=\"navigation\" aria-label=\"' . JText::_('JLIB_HTML_PAGINATION') . '\" class=\"pagination\">\n";
	$html .= $list['pageslinks'];
	$html .= "\n<input type=\"hidden\" name=\"" . $list['prefix'] . "limitstart\" value=\"" . $list['limitstart'] . "\" />";
	$html .= "\n</nav>";

	return $html;
}
````
2.  It seems to me that directing, especially at my address, and not for the first time, the comments such as: " I am not surprised that the a11y team once again chose to do nothing and instead criticise those that actually contribute to making joomla more accessible" is at least out of place.

3. As I wrote earlier - if you leave the code unchanged, also  I will test it successfully.

**Comment by brianteeman** [comment]
your code will break the templates i will not be changing it

**Comment by zwiastunsw** [comment]
OK.

**Comment by zwiastunsw** [comment]
I have tested this item :white_check_mark: successfully on fa05cc6497dedfd33ea166ec9d436d84423cc032<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18326">issues.joomla.org/tracker/joomla-cms/18326</a>.</sub>

**Comment by ghost** [comment]
RTC after 3 successful tests.

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**